Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm

```diff
@@ -844,18 +844,24 @@ static inline float adjustedUnexposedMaxEdge(float documentEdge, float exposedRe
     return WebCore::screenSize();
 }
 
+static FloatSize fullscreenPreferencesScreenSize(CGFloat preferredWidth)
+{
+    CGFloat preferredHeight = preferredWidth / kTargetFullscreenAspectRatio;
+    return FloatSize(CGSizeMake(preferredWidth, preferredHeight));
+}
+
 FloatSize WebPageProxy::availableScreenSize()
 {
+#if HAVE(UIKIT_WEBKIT_INTERNALS)
+    return fullscreenPreferencesScreenSize(m_preferences->mediaPreferredFullscreenWidth());
```

Comment:
This results in consecutive returns, which is a bit weird. Maybe you want an #else?